src/graphql.js

```diff
@@ -0,0 +1,142 @@
+        buyerCountry,
+        currency,
+        commit,
+        vault,
+        intent:         intent ? intent.toUpperCase() : intent,
+        disableFunding: disableFunding ? disableFunding.map(f => f && f.toUpperCase()) : disableFunding,
+        disableCard:    disableCard ? disableCard.map(f => f && f.toUpperCase()) : disableCard
+    };
+}
+
+export function callGraphQL<T, V>({ query, variables = {}, headers = {} } : { query : string, variables : V, headers? : { [string] : string } }) : ZalgoPromise<T> {
+    return request({
+        url:     GRAPHQL_URI,
+        method:  'POST',
+        json:    {
+            query,
+            variables
+        },
+        headers: {
+            'x-app-name': 'hosted_fields',
+            ...headers
+        }
+    }).then(({ status, body }) => {
+        const errors = body.errors || [];
+
+        if (errors.length) {
+            const message = errors[0].message || JSON.stringify(errors[0]);
+            throw new Error(message);
+        }
+
+        if (status !== 200) {
+            throw new Error(`${ GRAPHQL_URI } returned status ${ status }`);
+        }
+
+        return body.data;
+    });
+}
+
+// call gql with multiple merchant ids to get fundingEligibility for card
+export function getCardEligibility() : ZalgoPromise<CardEligibility> {
+    const variables = buildFundingEligibilityVariables();
+
+    try {
+        return callGraphQL({
+            query: `
+                query GetFundingEligibility(
+                    $clientID:String,
+                    $merchantID:[ String ],
+                    $buyerCountry:CountryCodes,
+                    $currency:SupportedCountryCurrencies,
+                    $intent:FundingEligibilityIntent,
+                    $commit:Boolean,
+                    $vault:Boolean,
+                    $disableFunding:[ SupportedPaymentMethodsType ],
+                    $disableCard:[ SupportedCardsType ]
+                ) {
+                fundingEligibility(
+                    clientId:$clientID,
+                    buyerCountry:$buyerCountry,
+                    currency:$currency,
+                    intent:$intent,
+                    commit:$commit,
+                    vault:$vault,
+                    disableFunding:$disableFunding,
+                    disableCard:$disableCard,
+                    merchantId:$merchantID
+                ) {
+                    card {
+                        eligible 
+                        branded 
+                    }
+                }
+              }
+            `,
+            variables
+        }).then((gqlResult) => {
+            if (!gqlResult || !gqlResult.fundingEligibility) {
+                throw new Error(`GraphQL fundingEligibility returned no fundingEligibility object`);
+            }
+            return gqlResult && gqlResult.fundingEligibility;
+        });
+    }
+    catch (err) {
+        getLogger().error(`GraphQL fundingEligibility error`, err);
```

Comment:
Let's do `graphql_fundingeligibility_error` -- error name needs to be a cal event, and payload should be an object like `{ err: stringifyError(err) }`